tests/performance/metrics/metrics.go

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package metrics
+
+import (
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	kvv1 "kubevirt.io/client-go/api/v1"
+)
+
+type MetricType int
+
+type Unit string
+
+// Metric is a struct for metrics for performance tests, it can be latencies or resource usage.
+type Metric struct {
+	Perc50 float64 `yaml:"perc50"`
```

Comment:
I would suggest making this a map: `Perc map[int]float64`. Adding or removing fields could be disruptive in the future. This would make this more general and we could easily add or remove values whenever we would want.

We could also have something like `Percentiles := []int{50, 90, 99}` which would be a static array of the percentiles that we would like to collect. Then setting `Metric.Perc` map values would be just a simple as:
```golang
for _, percentile := Percentiles {
  metric.Perc[percentile] = data[int(math.Ceil(float64(length*percentile)/100))-1].Value
}
```

Reply:
It makes sense! Thanks